src/app/rom.cc

```diff
@@ -25,77 +25,52 @@ namespace app {
 
 namespace {
 
-uint GetGraphicsAddress(const uchar* data, uint8_t offset) {
-  auto snes_address =
-      (uint)((((data[0x4F80 + offset]) << 16) | ((data[0x505F + offset]) << 8) |
-              ((data[0x513E + offset]))));
-  return core::SnesToPc(snes_address);
+int GetGraphicsAddress(const uchar* data, uint8_t offset) {
+  auto part_one = data[0x4F80 + offset] << 16;
+  auto part_two = data[0x505F + offset] << 8;
+  auto part_three = data[0x513E + offset];
+  auto snes_addr = (part_one | part_two | part_three);
+  return core::SnesToPc(snes_addr);
 }
 
-char* HexString(const char* str, const uint size) {
-  char* toret = (char*)malloc(size * 3 + 1);
-
-  uint i;
-  for (i = 0; i < size; i++) {
-    sprintf(toret + i * 3, "%02X ", (unsigned char)str[i]);
-  }
-  toret[size * 3] = 0;
-  return toret;
-}
-
-void PrintCompressionPiece(CompressionPiece* piece) {
+void PrintCompressionPiece(const std::shared_ptr<CompressionPiece>& piece) {
   printf("Command : %d\n", piece->command);
   printf("length  : %d\n", piece->length);
-  printf("Argument length : %d\n", piece->argument_length);
-  printf("Argument :%s\n", HexString(piece->argument, piece->argument_length));
-}
-
-CompressionPiece* NewCompressionPiece(const char command, const int length,
-                                      const char* args,
-                                      const int argument_length) {
-  CompressionPiece* toret = (CompressionPiece*)malloc(sizeof(CompressionPiece));
-  toret->command = command;
-  toret->length = length;
-  if (args != nullptr) {
-    toret->argument = (char*)malloc(argument_length);
-    memcpy(toret->argument, args, argument_length);
-  } else
-    toret->argument = nullptr;
-  toret->argument_length = argument_length;
-  toret->next = nullptr;
-  return toret;
-}
-
-void FreeCompressionPiece(CompressionPiece* piece) {
-  free(piece->argument);
-  free(piece);
+  printf("Argument :");
+  for (int i = 0; i < piece->argument.size(); ++i) {
```

Comment:
store `piece->argument.size()` in a cache variable for optimization

Reply:
Thanks Mitchell!